### PR TITLE
Add comprehensive unit and integration tests for Keycloak functionality

### PR DESCRIPTION
**Pull request by jkyberneees**
- Implement unit tests for the Token class, covering constructor behavior, expiration checks, role verification, and edge cases.
- Create integration tests for Keycloak interactions, including access token generation, role verification, token verification (both online and offline), and error handling.
- Set up Docker-based integration testing environment with Keycloak v26 and PostgreSQL.
- Include a realm export configuration for testing, defining users, roles, and client settings.
- Add a script to wait for Keycloak to be ready before running integration tests.
- Document integration test setup and usage in README.md.

### CODE REVIEW

File: libs/Jwt.ts [hunk 1]
```diff
@@ -1,31 +1,103 @@
...
+   * Defaults to `RS256` algorithm allowed list for safety.
+   *
+   * @param accessToken - JWT string to be verified
+   * @param cert - Public certificate or key used for verification
+   * @param options - Optional jsonwebtoken VerifyOptions
+   * @returns A Promise resolving to a `Token` instance if verification succeeds
+   * @throws {Error} When verification fails (signature mismatch or invalid token)
+   * @example
+   * const token = await jwt.verifyOffline('ey...', pubKey)
+   */
```
[the remark this answers is not in the document]
Comment (jkyberneees) [reply libs/Jwt.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/BackendStack21/keycloak-backend/pull/20#discussion_r2554116135)

File: libs/Jwt.ts [hunk 2]
```diff
@@ -1,31 +1,103 @@
...
+  /**
+   * Decode a token into a `Token` wrapper without performing cryptographic
+   * verification. Useful in contexts where the token will be inspected
+   * but not trusted until verified by other means.
+   *
+   * @param accessToken - The JWT string to decode
+   * @returns A `Token` instance containing the parsed payload
+   * @example
+   * const token = jwt.decode('ey...')
+   */
```
[the remark this answers is not in the document]
Comment (jkyberneees) [reply libs/Jwt.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/BackendStack21/keycloak-backend/pull/20#discussion_r2554116138)

File: libs/Jwt.ts [hunk 3]
```diff
@@ -1,31 +1,103 @@
...
+   * to make sure the token is still valid on the server-side. If the
+   * call completes successfully the token is accepted and returned as a
+   * `Token` wrapper for callers to inspect claims.
+   *
+   * @param accessToken - The JWT string to verify via Keycloak server
+   * @returns A Promise resolving to a `Token` instance when userinfo succeeds
+   * @throws {AxiosError} When the `userinfo` endpoint returns a non-2xx response
+   * @example
+   * const token = await jwt.verify('ey...')
+   */
```
[the remark this answers is not in the document]
Comment (jkyberneees) [reply libs/Jwt.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/BackendStack21/keycloak-backend/pull/20#discussion_r2554116140)
